### PR TITLE
Don't use mocks in production, use Engine if configured

### PR DESCRIPTION
**Pull request by nebolsin**
- [x] ~use Apollo Engine if configured~
- [x] disable mocks in non-dev environment

### CODE REVIEW

File: src/graphql.ts [hunk 1]
```diff
@@ -6,13 +6,16 @@ import { BIND_ADDRESS, PORT } from "./common/util/secrets";
...
 const server = new ApolloServer({
   schema,
   tracing: true,
-  mocks: true,
   introspection: true,
   playground: true,
-  debug: process.env.NODE_ENV === "development",
+  mocks: isDev,
+  debug: isDev,
+  engine: process.env.ENGINE_API_KEY !== undefined,
```
Comment (gzigzigzeo) [inline src/graphql.ts]: engine: true?

Comment (nebolsin) [reply src/graphql.ts]: Yes, we want `engine: true` if ENGINE_API_KEY is set

Comment (gzigzigzeo) [reply src/graphql.ts]: From what I see in main Apollo Engine example:
```
engine: {
    apiKey: process.env.API_KEY
  }
```

What am I missing?

Comment (nebolsin) [reply src/graphql.ts]: Just thought maybe also check it’s not an empty string...

Comment (gzigzigzeo) [reply src/graphql.ts]: Are you sure `engine: true` will work? I see different ways of engine configuration in docs, all of them require explicit passage of api key:

https://www.apollographql.com/docs/engine/setup-node.html

Comment (nebolsin) [reply src/graphql.ts]: https://www.apollographql.com/docs/apollo-server/v2/migration-engine.html#Stand-alone-Apollo-Server
In 2.0 it autodetects the env var. Hmm, maybe we don’t even need this flag.

Comment (nebolsin) [reply src/graphql.ts]: Yes, I’m pretty sure `true` will work: https://www.apollographql.com/docs/apollo-server/v2/api/apollo-server.html. But maybe we don’t even need it

Comment (nebolsin) [reply src/graphql.ts]: Turned out to be unnecessary, Engine is available now (mobius-network org members should have access): https://engine.apollographql.com/service/astrograph/. Corresponding infra changes have been pushed.

**Pull request merged**